File: backend/app/schemas/theme.py
"""
Pydantic schemas for Theme.
"""
from pydantic import BaseModel, ConfigDict


class ThemeBase(BaseModel):
    name: str


class ThemeCreate(ThemeBase):
    pass


class ThemeOut(ThemeBase):
    model_config = ConfigDict(from_attributes=True)
    id: str


class AssignThemesRequest(BaseModel):
    theme_ids: list[str]

File: frontend/src/types/theme.ts
/**
 * Mirrors backend: app/models/theme.py, app/schemas/theme.py
 */

export interface Theme {
  id: string;
  name: string;
}

export interface ThemeCreatePayload {
  name: string;
}

export interface AssignThemesPayload {
  theme_ids: string[];
}
